store/tikv/scan.go

```diff
@@ -199,7 +202,7 @@ func (s *Scanner) getData(bo *Backoffer) error {
 			// No more data in current Region. Next getData() starts
 			// from current Region's endKey.
 			s.nextStartKey = loc.EndKey
-			if len(loc.EndKey) == 0 {
+			if len(loc.EndKey) == 0 || (len(s.endKey) > 0 && kv.Key(s.nextStartKey).Cmp(kv.Key(s.endKey)) >= 0) {
```

Comment:
I think you want to check `endKey` in `Scanner.Next()` too.

Reply:
It makes sense.

Reply:
Then TiDB can't work with older TiKV normally. Do you think that's ok?

Reply:
Good point. I'd like to say it is acceptable -- it won't break anything because current usages do not rely on the iterator to stop at `upperBound`. But it could be more safe to check bound in `Next`.